ui/app/AppLayouts/Profile/views/MessagingView.qml

```diff
@@ -220,24 +248,17 @@ SettingsContentBase {
                 // TODO find a better icon for this
                 asset.name: Style.svg('globe')
                 asset.isImage: true
-                Component.onCompleted: {
-                    if (localAccountSensitiveSettings.displayChatImages) {
-                        showMessageLinksSwitch.checked = true
-                    }
-                }
                 components: [
                     StatusSwitch {
                         id: imageSwitch
                         checked: localAccountSensitiveSettings.displayChatImages
-                        onCheckedChanged: {
-                            if (localAccountSensitiveSettings.displayChatImages !== checked) {
-                                localAccountSensitiveSettings.displayChatImages = checked
-                            }
+                        onToggled: {
+                                localAccountSensitiveSettings.displayChatImages = !localAccountSensitiveSettings.displayChatImages
                         }
                     }
                 ]
                 onClicked: {
-                    imageSwitch.checked = !imageSwitch.checked
+                    localAccountSensitiveSettings.displayChatImages = !localAccountSensitiveSettings.displayChatImages
```

Comment:
Just toggle the switch instead of repeating the same code

Reply:
apparently switch.toggle() won't trigger onToggled. OnToggled will be triggered only on user input.

Reply:
Ah, that's stupid :/